pageserver/src/tenant/tasks.rs

```diff
@@ -183,7 +183,16 @@ async fn compaction_loop(tenant: Arc<Tenant>, cancel: CancellationToken) {
                 }
             };
 
-            warn_when_period_overrun(started_at.elapsed(), period, BackgroundLoopKind::Compaction);
+            if !first {
```

Comment:
first is never true here. I don't recall why I added it into the loop.

Reply:
Maybe it's easier to use a `for round in 0..` instead of loop for this?

Reply:
Also recalled that @problame has some dashboards for tracking these. This will mess up that data, but it probably does not matter.

Reply:
oops, good catch.  Moved the `first=false` to the end of the loop (there are no `continue` statements, so end of loop should always be hit before next iteration)

Reply:
I'm always scared of later introductions of `continue`, but let's me vigilant on following PRs.